### PR TITLE
Navigation Component, Functionality

### PR DESCRIPTION
**Pull request by andershopkins**
Componentized Nav, broke into partials, restructured, styled it. Updated SearchBox, added QuickStartMenu partial without functionality. Updated TW configs.

**Note**: There is still some work to be done to address a11y/keyboard navigation concerns, I've created a Trello card to handle this separately.

### CODE REVIEW

File: components/searchBox.tsx [hunk 1]
```diff
@@ -1,13 +1,55 @@
...
+        <span aria-hidden="true" className={classnames('absolute', 'left-4', 'z-20')}>
+          <svg
+            width="16px"
+            height="16px"
+            viewBox="0 0 16 16"
+            version="1.1"
+            xmlns="http://www.w3.org/2000/svg"
+          >
+            <title>icon-search</title>
+            <g id="icon-search" fill="currentColor" fillRule="nonzero">
```
Comment (bhgeorge) [inline components/searchBox.tsx:25]: Unsure if we need the title?

Also the id here and in the path below (Icon-Search) can probably be removed. Could probably remove the `<g>` too

File: components/site/Nav/Nav.tsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+        'px-gutter',
+        'shadow'
+      )}
+    >
+      <Link href="/">
+        <a className={classnames('flex', 'items-center', 'mr-8', 'flex-shrink-0')}>
+          <span className={classnames('block', 'relative', 'w-36', 'h-9', 'lg:w-48', 'lg:h-12')}>
+            <Image src="/sitecore.svg" layout="fill" />
+          </span>
+          {/* <Image src="/sitecore.svg" height="33" width="150" className={classnames('block')}/> */}
```
Comment (bhgeorge) [inline components/site/Nav/Nav.tsx:60]: Delete?

File: tailwindcss-custom-types.ts [hunk 3]
```diff
@@ -1,9 +1,18 @@
...
 type SCDPTailwindTypes =
+  | 'px-gutter-all'
   | 'heading-xl'
   | 'heading-lg'
   | 'heading-md'
   | 'heading-sm'
   | 'heading-xs'
+  | NavTWClasses
+  // @TODO - some transform utilities that are enabled by default
+  // aren't being added to the classname gen. Why?
```
Comment (bhgeorge) [inline tailwindcss-custom-types.ts:12]: Yes. I don't  know why this is. Noticed it in HZTL too. Maybe a bug w/ TW Classnames?

**Review by bhgeorge — CHANGES_REQUESTED** [review]

File: tailwindcss-custom-types.ts (hunk 3 above)
Comment (andershopkins) [reply tailwindcss-custom-types.ts:12]: Probably. I created a Trello card for one of us to investigate/open up an issue on TW Classnames when we have a little bit of time.

Comment (andershopkins) [reply tailwindcss-custom-types.ts:12]: https://trello.com/c/4KggUhXG/43-open-issue-for-tailwind-classnames-transform-utilities-not-generated

**Comment by andershopkins** [comment]
@bhgeorge  - updated and resolved all your requested changes.

**Review by bhgeorge — APPROVED** [review]

**Pull request merged**